pkgs/servers/web-apps/plausible/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib, stdenv, beamPackages, fetchFromGitHub, glibcLocales, cacert
+, mkYarnModules, nodejs, fetchpatch, nixosTests
+}:
+
+let
+  pname = "plausible";
+  version = "1.3.0";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "plausible";
+    repo = "analytics";
+    rev = "v${version}";
+    sha256 = "03lm1f29gwwixnhgjish5bhi3m73qyp71ns2sczdnwnbhrw61zps";
+  };
+
+  mixDeps = beamPackages.fetchMixDeps {
```

Comment:
Not sure how you feel about the FOD vs Nix dependencies thing.
mix2nix has just been added to nixpkgs, and the PR to enable mix2nix generated dependencies to be passed to mixRelease, should be merged during the course of next week.
I'm just mentioning this in case you are interested.

Reply:
> Not sure how you feel about the FOD vs Nix dependencies thing.

I dislike it, but I use it when it's more practical :sweat_smile: 

In this case I simply wasn't aware of `mix2nix` since I wrote most of it back in Februrary (where most of the current tooling didn't even exist yet AFAIU). What's your experience with it? If it doesn't create additional workload, I'm happy to change this :)

Reply:
I started testing it a couple of weeks ago, so basically take my opinion with a grain of salt.
However so far, I like it. I'm actually interested in figuring out the rough edges, so if you are interested, I can help with any problem you encounter with it. I worked with the person who did the PR to understand properly was was happening and potential failures. I'm by no means a mix expert, but I'm fairly confident we could get this to work should you encounter any problem. So if you like the nix approach better, I would say go for it!
I tested on plausible and it does generate a 1000 lines deps.nix file though. I think those are pretty standard dependencies, so not a problem, but if you don't want to bother with this, it makes sense too.

Reply:
Here is what it would look like
```
{ lib, beamPackages, overrides ? (x: y: {}) }:

let
	buildRebar3 = lib.makeOverridable beamPackages.buildRebar3;
	buildMix = lib.makeOverridable beamPackages.buildMix;
	buildErlangMk = lib.makeOverridable beamPackages.buildErlangMk;

	self = packages // (overrides self packages);

	packages = with beamPackages; with self; {
		ecto_sql = buildMix rec {
			name = "ecto_sql";
			version = "3.6.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0y871vj9s54dmg8snqhdcml6p7ccmd8n67mxzszvcy2r5lzmrwv6";
			};

			beamDeps = [ db_connection ecto postgrex telemetry ];
		};

		ink = buildMix rec {
			name = "ink";
			version = "1.2.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0wilhwd65x03fd6saq8arwdjg3mdq1ij8ljwgrb973k9lnh0qmr6";
			};

			beamDeps = [ jason ];
		};

		ssl_verify_fun = buildRebar3 rec {
			name = "ssl_verify_fun";
			version = "1.1.6";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1026l1z1jh25z8bfrhaw0ryk5gprhrpnirq877zqhg253x3x5c5x";
			};

			beamDeps = [];
		};

		oauther = buildMix rec {
			name = "oauther";
			version = "1.1.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "13kqwl35zard67qjpcn3d6yl6dl9fnwpxifdris1hcj540qg8x4k";
			};

			beamDeps = [];
		};

		phoenix_html = buildMix rec {
			name = "phoenix_html";
			version = "2.14.3";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "03z8r285znlg25yi47d4l59s7jq58y4dnhvbxgp16npkzykrgmpg";
			};

			beamDeps = [ plug ];
		};

		phoenix_ecto = buildMix rec {
			name = "phoenix_ecto";
			version = "4.2.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1h23fy3pylaszh3l2zafq1a7fjwlwb3yw7dy09p0mb4wi6p1p2j7";
			};

			beamDeps = [ ecto phoenix_html plug ];
		};

		phoenix_pubsub = buildMix rec {
			name = "phoenix_pubsub";
			version = "2.0.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0wgpa19l6xar0k2m117iz2kq3cw433llp07sqswpf5969y698bf5";
			};

			beamDeps = [];
		};

		poison = buildMix rec {
			name = "poison";
			version = "3.1.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1kng8xadrs03i77irxvdk9vfncrqzncmgxc5gc8y8gkknw76dj7y";
			};

			beamDeps = [];
		};

		certifi = buildRebar3 rec {
			name = "certifi";
			version = "2.6.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0zmvagzisnk7lj5pfipl19mjq9wn70i339hpbkfljf0vk6s9fk2j";
			};

			beamDeps = [];
		};

		siphash = buildMix rec {
			name = "siphash";
			version = "3.2.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1hxpn31m9w1xyfw8za4v0wxpq24r9256w62yfix675bf3iq10f5s";
			};

			beamDeps = [];
		};

		phoenix = buildMix rec {
			name = "phoenix";
			version = "1.5.8";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0mfr691wb95ym4jj2jfrph0x5k1fha43phxnga61cdj85yix1pim";
			};

			beamDeps = [ jason phoenix_html phoenix_pubsub plug plug_cowboy plug_crypto telemetry ];
		};

		appsignal_plug = buildMix rec {
			name = "appsignal_plug";
			version = "2.0.4";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1b434bcq5lnrjmdjin9nqg5sxi9b3na0x5cn7i0j5gi8dwgb0i56";
			};

			beamDeps = [ appsignal plug ];
		};

		telemetry = buildRebar3 rec {
			name = "telemetry";
			version = "0.4.3";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0hc0fr2bh97wah9ycpm7hb5jdqr5hnl1s3b2ibbbx9gxbwvbhwpb";
			};

			beamDeps = [];
		};

		erlex = buildMix rec {
			name = "erlex";
			version = "0.2.6";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0x8c1j62y748ldvlh46sxzv5514rpzm809vxn594vd7y25by5lif";
			};

			beamDeps = [];
		};

		nanoid = buildMix rec {
			name = "nanoid";
			version = "2.0.5";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0n2z7pjkg2ksivn56ryj9is6pcr6zwwhaxs8m9rdl10i69v8hvlm";
			};

			beamDeps = [];
		};

		cowboy_telemetry = buildRebar3 rec {
			name = "cowboy_telemetry";
			version = "0.3.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1bzhcdq12p837cii2jgvzjyrffiwgm5bsb1pra2an3hkcqrzsvis";
			};

			beamDeps = [ cowboy telemetry ];
		};

		geolix = buildMix rec {
			name = "geolix";
			version = "1.1.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0033n5zzd4lsvrsippjzab209mq6g0kcb1lyqy6jh37kmvr5824q";
			};

			beamDeps = [ poolboy ];
		};

		postgrex = buildMix rec {
			name = "postgrex";
			version = "0.15.9";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1vmd63vxwz8knid424b0rbp200vj7q7rz3xp98yj5cjc7q81j1v1";
			};

			beamDeps = [ connection db_connection decimal jason ];
		};

		parallel_stream = buildMix rec {
			name = "parallel_stream";
			version = "1.0.6";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1sr3127izwy40j6sr25j745if70nblravwj2xfwqf6z1963jx6v3";
			};

			beamDeps = [];
		};

		excoveralls = buildMix rec {
			name = "excoveralls";
			version = "0.13.4";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "14313qwlvcj4yv3sglhph9r45ycigjd6c2y1881xyp73xsrh1bps";
			};

			beamDeps = [ hackney jason ];
		};

		php_serializer = buildMix rec {
			name = "php_serializer";
			version = "0.9.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "16m5ghdb74mxrnf9w31rq25cyz9fq1iv6wv747ygfi098rd87srl";
			};

			beamDeps = [];
		};

		bamboo_phoenix = buildMix rec {
			name = "bamboo_phoenix";
			version = "1.0.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1vkxnfw4xmaz7339nmbcrqlpr260g7c2pfwlg6j897014sxqzf3d";
			};

			beamDeps = [ bamboo phoenix ];
		};

		jumper = buildMix rec {
			name = "jumper";
			version = "1.0.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0cvlbfkapkvbwaijmjq3cxg5m6yv4rh69wvss9kfj862i83mk31i";
			};

			beamDeps = [];
		};

		junit_formatter = buildMix rec {
			name = "junit_formatter";
			version = "3.1.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1x3j7gbv2sk0mvjgrxcnbnnzq5lsiykdvazzfx7zq4gpjcd40lns";
			};

			beamDeps = [];
		};

		csv = buildMix rec {
			name = "csv";
			version = "2.4.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0ychghqb3smkxy42dz6pcpb9anddwc39dx0fn5k7kqk7mhw8jl2l";
			};

			beamDeps = [ parallel_stream ];
		};

		httpoison = buildMix rec {
			name = "httpoison";
			version = "1.8.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0fiwkdrbj7mmz449skp7laz2jdwsqn3svddncmicd46gk2m9w218";
			};

			beamDeps = [ hackney ];
		};

		envy = buildMix rec {
			name = "envy";
			version = "1.1.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "129s5hjw957i12qmq9k094s4h7hbvh8fr3ax2ibxfps18wdfnqbh";
			};

			beamDeps = [];
		};

		decimal = buildMix rec {
			name = "decimal";
			version = "1.9.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "11bxdzywr4pam7l0jpfrknhzlnz9zpgz473m7s7r5mpfd0sk9wmi";
			};

			beamDeps = [];
		};

		db_connection = buildMix rec {
			name = "db_connection";
			version = "2.2.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0307z0v1awc3scjvh955w4kw7gdw5g6nynmsnj9zpbx8v10g4ak4";
			};

			beamDeps = [ connection ];
		};

		idna = buildRebar3 rec {
			name = "idna";
			version = "6.1.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1sjcjibl34sprpf1dgdmzfww24xlyy34lpj7mhcys4j4i6vnwdwj";
			};

			beamDeps = [ unicode_util_compat ];
		};

		sshex = buildMix rec {
			name = "sshex";
			version = "2.2.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1nqkgf4ymcmgwzrvdsh3p6yfi8vq4svf6xrd1bkm1j0x06jwmcj5";
			};

			beamDeps = [];
		};

		credo = buildMix rec {
			name = "credo";
			version = "1.5.4";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1i0v2ivh4m8kvh9kllxpyws1pj8mqkd6zd8kp8wky2nwx92sylfg";
			};

			beamDeps = [ bunt file_system jason ];
		};

		decorator = buildMix rec {
			name = "decorator";
			version = "1.3.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1z57s5x4f2d24kj32lzj9bcs15qmgcqfskm8krnrwmy8wf4x02xq";
			};

			beamDeps = [];
		};

		parse_trans = buildRebar3 rec {
			name = "parse_trans";
			version = "3.3.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "12w8ai6b5s6b4hnvkav7hwxd846zdd74r32f84nkcmjzi1vrbk87";
			};

			beamDeps = [];
		};

		cachex = buildMix rec {
			name = "cachex";
			version = "3.3.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "189irin4xkbnj6b3ih1h5fvli1xq6m1sz1xiyqryyk71vphmw3nr";
			};

			beamDeps = [ eternal jumper sleeplocks unsafe ];
		};

		kaffy = buildMix rec {
			name = "kaffy";
			version = "0.9.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1lvf9h6pvzyyyn53axaf0g3snmyk1i8p27nimqrv9zk8irxzb3yi";
			};

			beamDeps = [ ecto phoenix phoenix_html ];
		};

		tzdata = buildMix rec {
			name = "tzdata";
			version = "1.0.5";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0nia83zpk0pb4jkpvhkmmgw8i5p6kd6cf776q6aj0pcym6i9llam";
			};

			beamDeps = [ hackney ];
		};

		elixir_uuid = buildMix rec {
			name = "elixir_uuid";
			version = "1.2.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0ljpb1l9v262k043irj6i6w9fwyqn0b2fj86jyhcwm9mdkma5szp";
			};

			beamDeps = [];
		};

		jason = buildMix rec {
			name = "jason";
			version = "1.1.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1zispkj3s923izkwkj2xvaxicd7m0vi2xnhnvvhkl82qm2y47y7x";
			};

			beamDeps = [ decimal ];
		};

		dialyxir = buildMix rec {
			name = "dialyxir";
			version = "1.0.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0m26cpw56gq1q47x1w7p5jy3sxaj5la1l1nq13519b2z2j46bc5f";
			};

			beamDeps = [ erlex ];
		};

		timex = buildMix rec {
			name = "timex";
			version = "3.6.3";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1v6rn17llandjdm8m3gdc2ylardsqn9czcww58888mngbzwz8sbd";
			};

			beamDeps = [ combine gettext tzdata ];
		};

		mime = buildMix rec {
			name = "mime";
			version = "1.6.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "19qrpnmaf3w8bblvkv6z5g82hzd10rhc7bqxvqyi88c37xhsi89i";
			};

			beamDeps = [];
		};

		unicode_util_compat = buildRebar3 rec {
			name = "unicode_util_compat";
			version = "0.7.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "08952lw8cjdw8w171lv8wqbrxc4rcmb3jhkrdb7n06gngpbfdvi5";
			};

			beamDeps = [];
		};

		cowlib = buildRebar3 rec {
			name = "cowlib";
			version = "2.9.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0w318ynls1iqvl5p8icyfbhysf12qvhr220narhrj3d78315s5z4";
			};

			beamDeps = [];
		};

		sentry = buildMix rec {
			name = "sentry";
			version = "8.0.5";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0z2bpm4j0x5liyw3d4575x7pc7s2vy54qs9ygdnqhbpmvfgq6wj9";
			};

			beamDeps = [ hackney jason plug plug_cowboy ];
		};

		phoenix_pagination = buildMix rec {
			name = "phoenix_pagination";
			version = "0.7.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0zw2dlfnfqgk0igm8dgldf8zp8m9f94y2whdpc1b6jw2x4w56197";
			};

			beamDeps = [ ecto_sql phoenix_html plug ];
		};

		comeonin = buildMix rec {
			name = "comeonin";
			version = "5.3.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "012zr4s7b5bipng6yszqxkqr1lcv7imf8gyvxad56jachh1396fh";
			};

			beamDeps = [];
		};

		yamerl = buildRebar3 rec {
			name = "yamerl";
			version = "0.8.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0gxjrrdrz9rcfmkg7x7z9p0fcz90dzqyk4laxz8gwi23svwk1jwn";
			};

			beamDeps = [];
		};

		ex_machina = buildMix rec {
			name = "ex_machina";
			version = "2.5.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1160arhhvvj0pk2dh8gs1lp6m9qyh3jg2d8kn6nziydk68f8a94g";
			};

			beamDeps = [ ecto ecto_sql ];
		};

		elixir_make = buildMix rec {
			name = "elixir_make";
			version = "0.6.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1na8agkks1hrwq1lxfj4yd96bvfcs4hk7mbra9z6lli2vanrxr03";
			};

			beamDeps = [];
		};

		double = buildMix rec {
			name = "double";
			version = "0.7.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0abk4nxq3bkxjkm63c6a5a5yy6n1xs7mj0xbggd54i3b4si8ghzh";
			};

			beamDeps = [];
		};

		cowboy = buildErlangMk rec {
			name = "cowboy";
			version = "2.8.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "12mcbyqyxjynzldcfm7kbpxb7l7swqyq0x9c2m6nvjaalzxy8hs6";
			};

			beamDeps = [ cowlib ranch ];
		};

		hackney = buildRebar3 rec {
			name = "hackney";
			version = "1.17.4";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "05kbk3rpw2j3cb9pybikydxmi2nm5pidpx0jsm48av2mjr4zy5ny";
			};

			beamDeps = [ certifi idna metrics mimerl parse_trans ssl_verify_fun unicode_util_compat ];
		};

		ref_inspector = buildMix rec {
			name = "ref_inspector";
			version = "1.3.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0m1q4fdy79hqrh7fp9a3l5pm8mqjd47gxqwnyxk9kip510dynwii";
			};

			beamDeps = [ hackney yamerl ];
		};

		gen_smtp = buildRebar3 rec {
			name = "gen_smtp";
			version = "1.1.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1dwszws49n7sza38js2999yzl3mn1yirxhyb9114mzby07651g2i";
			};

			beamDeps = [ hut ranch ];
		};

		metrics = buildRebar3 rec {
			name = "metrics";
			version = "1.0.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "05lz15piphyhvvm3d1ldjyw0zsrvz50d2m5f2q3s8x2gvkfrmc39";
			};

			beamDeps = [];
		};

		geolix_adapter_mmdb2 = buildMix rec {
			name = "geolix_adapter_mmdb2";
			version = "0.5.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0gndrrdr5paz1vgs0lw44pqxn62p4j5441wjji0yblx2l2v8a56b";
			};

			beamDeps = [ geolix mmdb2_decoder ];
		};

		sleeplocks = buildRebar3 rec {
			name = "sleeplocks";
			version = "1.1.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1q823i5bisc83pyssgrqkggyxiasm7b8dygzj2r943adzyp3gvl4";
			};

			beamDeps = [];
		};

		bcrypt_elixir = buildMix rec {
			name = "bcrypt_elixir";
			version = "2.3.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0s9psinz913l690xbcrl21m23zwinw4r2ypjgg7ybl3f9wfxd09c";
			};

			beamDeps = [ comeonin elixir_make ];
		};

		file_system = buildMix rec {
			name = "file_system";
			version = "0.2.10";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1p0myxmnjjds8bbg69dd6fvhk8q3n7lb78zd4qvmjajnzgdmw6a1";
			};

			beamDeps = [];
		};

		ua_inspector = buildMix rec {
			name = "ua_inspector";
			version = "0.20.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "09kz0bb3mj9w3mq8khlj870jwyg81gaszhigw65xbrsmkwxn5s1h";
			};

			beamDeps = [ hackney poolboy yamerl ];
		};

		combination = buildMix rec {
			name = "combination";
			version = "0.0.3";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1jhqvaggg55qvrlkz182b5n7pdbhf062a79pqryyyhnzcgs9kc3j";
			};

			beamDeps = [];
		};

		plug_crypto = buildMix rec {
			name = "plug_crypto";
			version = "1.2.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1nxnxj62iv4yvm4771jbxpj3l4brn2crz053y12s998lv5x1qqw7";
			};

			beamDeps = [];
		};

		cors_plug = buildMix rec {
			name = "cors_plug";
			version = "1.5.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0sfc0a5n8r4przz0pspyf5c9cqhcgqj8fsss6k2d0bf11p92gw4s";
			};

			beamDeps = [ plug ];
		};

		hut = buildRebar3 rec {
			name = "hut";
			version = "1.3.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0qxmkazkakrmvd9n1im4ad43wxgh189fqcd9lfsz58fqan2x45by";
			};

			beamDeps = [];
		};

		gettext = buildMix rec {
			name = "gettext";
			version = "0.18.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1igmn69xzj5wpkblg3k9v7wa2fjc2j0cncwx0grk1pag7nqkgxgr";
			};

			beamDeps = [];
		};

		poolboy = buildRebar3 rec {
			name = "poolboy";
			version = "1.5.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1qq116314418jp4skxg8c6jx29fwp688a738lgaz6h2lrq29gmys";
			};

			beamDeps = [];
		};

		connection = buildMix rec {
			name = "connection";
			version = "1.0.4";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1ng420jq6f0rk8k417i0bkaggxaiq0bsnw8ajbwkm912pv4m022a";
			};

			beamDeps = [];
		};

		bamboo = buildMix rec {
			name = "bamboo";
			version = "2.0.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0533pzfhqkh1haivc70s2pxm92f88gmk49x7790ikb6d9z7mg385";
			};

			beamDeps = [ hackney jason mime plug ];
		};

		unsafe = buildMix rec {
			name = "unsafe";
			version = "1.0.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1rahpgz1lsd66r7ycns1ryz2qymamz1anrlps986900lsai2jxvc";
			};

			beamDeps = [];
		};

		ranch = buildRebar3 rec {
			name = "ranch";
			version = "1.7.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1my7mz3x7a1fmjyin55nn1fr2d2rl3y64qf3kpcidxvxg0kqa7a5";
			};

			beamDeps = [];
		};

		appsignal_phoenix = buildMix rec {
			name = "appsignal_phoenix";
			version = "2.0.3";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "07wif63y62vh0i0vpyvisd6pya33ygb0qfgpmgngfwlv997yksii";
			};

			beamDeps = [ appsignal_plug phoenix phoenix_html telemetry ];
		};

		combine = buildMix rec {
			name = "combine";
			version = "0.10.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "06s5y8b0snr1s5ax9v3s7rc6c8xf5vj6878d1mc7cc07j0bvq78v";
			};

			beamDeps = [];
		};

		oban = buildMix rec {
			name = "oban";
			version = "2.6.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1sjllaa0ir1rwa6hdclyf2b9dlaj8fnldzbw9zv1l76xiymym1iy";
			};

			beamDeps = [ ecto_sql jason postgrex telemetry ];
		};

		mix_test_watch = buildMix rec {
			name = "mix_test_watch";
			version = "1.0.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "01qs8kfbzz0rv9k7dsvsp6yhln8mri7x1ikj75vq9xh6if6mbb27";
			};

			beamDeps = [ file_system ];
		};

		phoenix_live_reload = buildMix rec {
			name = "phoenix_live_reload";
			version = "1.3.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "10mmjmpp278kvsmqnrw4g63pjwb13sdqc6hwqrpbgwlkwdx01h82";
			};

			beamDeps = [ file_system phoenix ];
		};

		browser = buildMix rec {
			name = "browser";
			version = "0.4.4";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "1mzqniwx1px25mg79y7nq4iv7g5b18wq00w75ds1jjsaklqclxnl";
			};

			beamDeps = [ plug ];
		};

		plug = buildMix rec {
			name = "plug";
			version = "1.11.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "07w83cx4xx90x4l1kmil4lpby55lpw83jfw3y08pqn5vxx7lwli3";
			};

			beamDeps = [ mime plug_crypto telemetry ];
		};

		bertex = buildMix rec {
			name = "bertex";
			version = "1.3.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0yp5iqyj477yl8hz3788igjh04j9rbhwldxfgdxlnxmmid3mwp8a";
			};

			beamDeps = [];
		};

		appsignal = buildMix rec {
			name = "appsignal";
			version = "2.0.8";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "092s07aj4ad81iybpc7m1sn3mvg2wqjdnisyyxrmhvp27qzrx7yi";
			};

			beamDeps = [ decorator hackney jason poison telemetry ];
		};

		mimerl = buildRebar3 rec {
			name = "mimerl";
			version = "1.2.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "08wkw73dy449n68ssrkz57gikfzqk3vfnf264s31jn5aa1b5hy7j";
			};

			beamDeps = [];
		};

		bamboo_smtp = buildMix rec {
			name = "bamboo_smtp";
			version = "4.0.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0ggjm05yf8psvvw98fijzfd5yp4fv6y59rpjycj9y6qjj98024i4";
			};

			beamDeps = [ bamboo gen_smtp ];
		};

		bunt = buildMix rec {
			name = "bunt";
			version = "0.2.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0lw3v9kwbbcy1v6ygziiky887gffwwmxvyg4r1v0zm71kzhcgxbs";
			};

			beamDeps = [];
		};

		plug_cowboy = buildMix rec {
			name = "plug_cowboy";
			version = "2.4.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "01q5ny2m75206987xgzjaakskqq8524nhnrakdypmczpvyv168fp";
			};

			beamDeps = [ cowboy cowboy_telemetry plug telemetry ];
		};

		mmdb2_decoder = buildMix rec {
			name = "mmdb2_decoder";
			version = "3.0.0";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "13pa1gr8c7pdw8i9vr32rrvcl790f9ndkxmrrqfqslqm54jck79m";
			};

			beamDeps = [];
		};

		ecto = buildMix rec {
			name = "ecto";
			version = "3.6.1";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "0lv5mypffqaxqkzhv78d9qvhdy4ckxs8lj150ygv2iq4k552kcyb";
			};

			beamDeps = [ decimal jason telemetry ];
		};

		eternal = buildMix rec {
			name = "eternal";
			version = "1.2.2";

			src = fetchHex {
				pkg = "${name}";
				version = "${version}";
				sha256 = "10p7m6kv2z2c16gw36wgiwnkykss4lfkmm71llxp09ipkhmy77rc";
			};

			beamDeps = [];
		};
	};
in self

```

Reply:
Just a heads up regarding mixDeps. I've merged the PR integrating nix dependencies into mixRelease. If you want to use nix dependencies the attribute you have to pass is mixNixDeps. You can continue to use the fod dependencies, you'll just have to change the name from `mixDeps` to `mixFodDeps`

Reply:
With `mixNIxDeps` generated & set I get the following compilation error:

```
plausible> == Compilation error in file lib/plausible/clickhouse_repo.ex ==
plausible> ** (ArgumentError) adapter ClickhouseEcto was not compiled, ensure it is correct and it is included as a project dependency
plausible>     (ecto 3.6.1) lib/ecto/repo/supervisor.ex:60: Ecto.Repo.Supervisor.compile_config/2
plausible>     lib/plausible/clickhouse_repo.ex:2: (module)
plausible>     (stdlib 3.14.2) erl_eval.erl:680: :erl_eval.do_apply/6
error: build of '/nix/store/283178iklq7nsbr9dqr443a0lklsjs4h-plausible-1.3.0.drv' on 'ssh-ng://builder-ng' failed: error: builder for '/nix/store/283178iklq7nsbr9dqr443a0lklsjs4h-plausible-1.3.0.drv' failed with exit code 1
error: builder for '/nix/store/283178iklq7nsbr9dqr443a0lklsjs4h-plausible-1.3.0.drv' failed with exit code 1

```

An idea what's going on here?

Reply:
Ah, the clickhouse dependency are git dependencies. mix2nix currently doesn't support git dependencendies. That's why you are seeing this missing dependency.
You'll have to use FOD until we figure out how to do the git dependencies properly

Reply:
Ok, thanks for the clarification! I'll resolve this for now as there's nothing actionable atm :)

Reply:
you just need to remember to change mixDeps to `mixFodDeps` that's it.